### DISCUSSION TITLE
How to provide interpretor?

### DISCUSSION THREAD

**Discussion by ikk6** [ndabAP/vue-command#493] [General]
Great project, thank you!
How can I provide my own interpreter for the component? Here's what I'm trying to do. I'm binding my custom function like that: `:interpretor="myFunc"`. It's getting called, the code is getting executed. The documentation is vague at this point - all it says is that I must use the [properties](https://github.com/ndabAP/vue-command#properties) and [exposed functions](https://github.com/ndabAP/vue-command#exposed). Can anyone please provide an example of what to call next to output anything and return to the prompt again? Because calling `createStdout` from `myFunc` has no effect.

**Comment by ndabAP** [comment]
Hey, thanks for your feedback.

For the moment, check what's happening after we call the interpreter: https://github.com/ndabAP/vue-command/blob/main/src/components/VueCommand.vue#L382. You have to mimic that and built in your desired behavior. 

I can give you a more detailed example in a few days.

**Comment by ikk6** [comment]
Okay, if I understand it correctly, everything you see on a screen is the history where every entry is a component. The query is `<VueCommandQuery>` and the output is `<VueCommandOut>`, no matter what it is - either a `<VueCommandStdout>` or a user-provided component wrapper as `<VueCommandOut>`. How exactly do I prepopulate the history object with the prompt and query? That would be nice to see some examples. Thank you.

**Comment by ndabAP** [comment]
You are right with all your statements.

Check the example for populating and mutating the history: https://github.com/ndabAP/vue-command/blob/main/src/hosted/App.vue#L18. You can pass an array as history property, which will synchronized by `vue-command` if you add the the `v-model` directive.

**Comment by ikk6** [comment]
While it's all cool, the problem I'm experiencing has to do with context. In my project, one of the commands is a vue component that uses `vue-command` itself. I.e. I'm trying to open another "terminal" as a command in a terminal. Now, when that nested terminal (which is basically a wrapper component around `vue-command`) calls for exposed methods (for instance, createStdout) those from the top-level component are being called with the nested component getting destroyed. That's confusing, shouldn't the context be scoped?

**Comment by ikk6** [comment]
Okay, it seems that tagging each instance of `vue-command` by `ref` and calling, for instance, `this.$refs.whatever.createStdout` helps to avoid confusion. This solves it for me, thank you.
